src/main/java/jewellery/inventory/service/UserService.java

```diff
@@ -31,23 +29,18 @@ public UserResponse getUser(UUID id) {
   }
 
   public UserResponse createUser(UserRequest user) {
-    if (checkIfEmailExists(user.getEmail(), null)) {
-      throw new DuplicateEmailException(user.getEmail());
-    }
-
-    if (checkIfNameExists(user.getName(), null)) {
-      throw new DuplicateNameException(user.getName());
-    }
-    return UserMapper.INSTANCE.toUserResponse(
-        userRepository.save(UserMapper.INSTANCE.toUserEntity(user)));
+    User userToCreate = UserMapper.INSTANCE.toUserEntity(user);
+    validateUserDetails(userToCreate);
+    return UserMapper.INSTANCE.toUserResponse(userRepository.save(userToCreate));
   }
 
   public UserResponse updateUser(UserRequest userRequest, UUID id) {
-    User existingUser =
-        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+    if (!userRepository.existsById(id)) {
+      throw new UserNotFoundException(id);
+    }
```

Comment:
Why did you need to change this? I am curious

Reply:
It is supposed to be more performant. With this we are just checking if the user exists while with the old version we checked and had the whole user 'extracted' which was unnecessary